pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -30,18 +30,19 @@ import (
 // This is necessitated by the Keyvault CSI implementation, which requires at least one mount
 // in order to start mirroring the Keyvault values into corresponding Kubernetes secret(s).
 type PlaceholderPodController struct {
-	client client.Client
-	config *config.Config
+	client     client.Client
+	config     *config.Config
+	ingConfigs []*manifests.NginxIngressConfig
 }
 
-func NewPlaceholderPodController(manager ctrl.Manager, conf *config.Config) error {
+func NewPlaceholderPodController(manager ctrl.Manager, conf *config.Config, ingConfigs []*manifests.NginxIngressConfig) error {
```

Comment:
Should we pass in an array of a more generic type instead of NginxIngressConfig, like an interface? Or is the plan to add additional configs (for example for traffic controller) via an additional parameter?

Reply:
This is something I was planning on doing. Was juggling lots of thoughts so it slipped my mind. Will swap to this real quick :)

Reply:
Added interfaces in a few places. Don't think we need one for the IngressReconciler (this feature will be phased out and we should only generate one Ingress per service so it doesn't make sense to generalize). The watchdog is pretty specific to nginx right now so can't generalize that either.